### ISSUE TITLE
Make PEM, PKCS5, PKCS12 work with Cipher light

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#8059]
- Change those modules from "depends on `CIPHER_C`" to "auto-enables `CIPHER_LIGHT` in `build-info.h`".
- Re-enable them in `all.sh` component `test_cipher_light_only` introduced by #8058.
- Remove the non-cipher implementation in `pem.c` that was temporarily kept in #8051.

If this is the first task that makes a module auto-enable `CIPHER_LIGHT`, also remove the temporary hack using `-D` in `CFLAGS` from `test_cipher_light_only`.

**Depends on:** #8058, #8051

**Comment by mpg** [comment]
Closing this issue as out of scope with respect to [the current goals](https://github.com/Mbed-TLS/mbedtls/issues/8337).
